controllers/tests/config/selfnoderemediationconfig_controller_test.go

```diff
@@ -343,3 +404,33 @@ func generateDs(name, namespace, version string) *appsv1.DaemonSet {
 		},
 	}
 }
+
+func verifySNRStatusRemoved(snr *selfnoderemediationv1alpha1.SelfNodeRemediation, statusType string) {
+	Eventually(func(g Gomega) {
+		tmpSNR := &selfnoderemediationv1alpha1.SelfNodeRemediation{}
+		g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(snr), tmpSNR)).To(Succeed())
+		g.Expect(meta.FindStatusCondition(tmpSNR.Status.Conditions, statusType)).To(BeNil())
+	}, 10*time.Second, 250*time.Millisecond).Should(Succeed())
+}
+
+func createSNR(snr *selfnoderemediationv1alpha1.SelfNodeRemediation) {
+	Expect(k8sClient.Create(context.Background(), snr)).To(Succeed())
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(snr), &selfnoderemediationv1alpha1.SelfNodeRemediation{})).To(Succeed())
+	}, 10*time.Second, 250*time.Millisecond).Should(Succeed())
+
+}
+
+func setSNRStatusDisabled(snr *selfnoderemediationv1alpha1.SelfNodeRemediation) {
+	snrTmp := &selfnoderemediationv1alpha1.SelfNodeRemediation{}
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(snr), snrTmp)).To(Succeed())
+		snrTmp.Status.Conditions = []metav1.Condition{metav1.Condition{
+			Type:               "Disabled",
+			Status:             metav1.ConditionTrue,
+			Reason:             "SNRDisabledConfigurationNotFound",
+			LastTransitionTime: metav1.Time{Time: time.Now()},
+		}}
+		g.Expect(k8sClient.Status().Update(context.Background(), snrTmp)).To(Succeed())
```

Comment:
Maybe we should verify that the update was persisted? Otherwise the test might return success before the CR even got the condition?